integration-tests/suites/mock_sensor/server.go

```diff
@@ -0,0 +1,460 @@
+	m.networkMutex.Lock()
+	defer m.networkMutex.Unlock()
+
+	m.logEvent("NetworkInfo: %s %s|%s|%s|%s|%s",
+		connection.GetContainerId(),
+		m.translateAddress(connection.GetLocalAddress()),
+		m.translateAddress(connection.GetRemoteAddress()),
+		connection.GetRole().String(),
+		connection.GetSocketFamily().String(),
+		connection.GetCloseTimestamp().String())
+
+	conn := types.NetworkInfo{
+		LocalAddress:   m.translateAddress(connection.LocalAddress),
+		RemoteAddress:  m.translateAddress(connection.RemoteAddress),
+		Role:           connection.GetRole().String(),
+		SocketFamily:   connection.GetSocketFamily().String(),
+		CloseTimestamp: connection.GetCloseTimestamp().String(),
+	}
+
+	if connections, ok := m.connections[containerID]; ok {
+		connections[conn] = true
+	} else {
+		connections := ConnMap{conn: true}
+		m.connections[containerID] = connections
+	}
+}
+
+// pushEndpoint converts an endpoint event into the test's own structure
+// and stores it
+func (m *MockSensor) pushEndpoint(containerID string, endpoint *sensorAPI.NetworkEndpoint) {
+	m.networkMutex.Lock()
+	defer m.networkMutex.Unlock()
+
+	m.logEvent("EndpointInfo: %s|%s|%s|%s|%s",
+		endpoint.GetSocketFamily().String(),
+		endpoint.GetProtocol().String(),
+		endpoint.GetListenAddress().String(),
+		endpoint.GetCloseTimestamp().String(),
+		endpoint.GetOriginator().String())
+
+	var originator types.ProcessOriginator
+	if endpoint.GetOriginator() != nil {
+		originator = types.ProcessOriginator{
+			ProcessName:         endpoint.GetOriginator().ProcessName,
+			ProcessArgs:         endpoint.GetOriginator().ProcessArgs,
+			ProcessExecFilePath: endpoint.GetOriginator().ProcessExecFilePath,
+		}
+	}
+
+	var listen types.ListenAddress
+	if endpoint.GetListenAddress() != nil {
+		listen = types.ListenAddress{
+			AddressData: string(endpoint.GetListenAddress().GetAddressData()),
+			Port:        int(endpoint.GetListenAddress().GetPort()),
+			IpNetwork:   string(endpoint.GetListenAddress().GetIpNetwork()),
+		}
+	}
+
+	ep := types.EndpointInfo{
+		Protocol:       endpoint.GetProtocol().String(),
+		Originator:     &originator,
+		CloseTimestamp: endpoint.GetCloseTimestamp().String(),
+		Address:        &listen,
+	}
+
+	if endpoints, ok := m.endpoints[containerID]; ok {
+		endpoints[ep] = true
+	} else {
+		endpoints := EndpointMap{ep: true}
+		m.endpoints[containerID] = endpoints
+	}
+}
+
+// translateAddress is a helper function for converting binary representations
+// of network addresses (in the signals) to usable forms for testing
+func (m *MockSensor) translateAddress(addr *sensorAPI.NetworkAddress) string {
+	ipPortPair := utils.NetworkPeerID{
+		Address: utils.IPFromBytes(addr.GetAddressData()),
+		Port:    uint16(addr.GetPort()),
+	}
+	return ipPortPair.String()
+}
+
+func (m *MockSensor) logEvent(format string, args ...interface{}) {
+	m.logFile.WriteString(fmt.Sprintf(format+"\n", args...))
+}
```

Comment:
I would assume go has some logging library we could use somewhere 🤔

Reply:
I originally avoided the log library because I didn't want to mess with logging elsewhere, but obviously Logger structs exist and I am idiot 🤣 I've implemented this locally now and it also gives us timestamps on the events which could make debugging of timing problems a lot easier